src/components/ids-month-view/ids-month-view.ts

```diff
@@ -1940,25 +1940,16 @@ class IdsMonthView extends Base {
     if (!this.container) return;
 
     const eventsContainer = this.container?.querySelector(`.events-container[data-key="${dateKey}"]`);
-    // const orders = [...(eventsContainer?.querySelectorAll('ids-calendar-event')) || []].map((elem: any) => elem.order);
     const orders = [...(eventsContainer?.childNodes) || []].map((elem: any) => elem.order);
     const baseOrder = orders.length ? Math.max(...orders) + 1 : 0;
     let isOverflowing = false;
-    const idsCalendar = document.querySelector('ids-calendar');
-    const isCustom = idsCalendar?.getAttribute('is-custom');
     events.forEach((event: CalendarEventData, index: number) => {
       const start = new Date(event.starts);
       const end = new Date(event.ends);
       const days = this.#countDays(start, end) || 1;
 
       for (let i = 0; i < days; i++) {
-        let calendarEvent = new IdsCalendarEvent();
-        // Custom Calendar Event
-        if (isCustom && customCalendarEvent?.name === 'MonthViewCalendarEventTemplate') {
-          if (customCalendarEvent.assignedNodes()[0]) {
-            calendarEvent = customCalendarEvent.assignedNodes()[0].cloneNode(true);
-          }
-        }
+        const { calendarEvent, isCustom } = this.#newCalendarEvent(customCalendarEvent);
```

Comment:
Moved to private function to hide logic and prevent the unnecessary creation of IdsCalendarEvent unless required.